### PR TITLE
Epoch description field

### LINKED ISSUE #1572: Epoch Meta Data or Notes
- People want to be able to write info about what the epoch is about.
- This would help add flavor to the epoch

In this design, we remove epoch naming such as epoch 1, epoch 2, etc., just allowing the date range to be the name. The user can now add a description to upcoming or current epochs. 

[UI](https://www.figma.com/file/L4oZwYVUdpgacZtwipGaYe/Coordinape-App?node-id=26295%3A76370)
[Prototype](https://www.figma.com/proto/L4oZwYVUdpgacZtwipGaYe/Coordinape-App?page-id=14369%3A30224&node-id=26295%3A75992&viewport=1239%2C18819%2C0.27&scaling=min-zoom&starting-point-node-id=14370%3A52369&show-proto-sidebar=1&hide-ui=1)

### PR DESCRIPTION
**Pull request by sshmm**
## Motivation and Context
resolves #1572

## Description
1- Add description column to epochs table
2- include description as optional argument for updateEpoch and createEpoch
3- display epoch description if available in the following places:
 -Overview /circles on the circle panel
- Epoch Overview /history for each epoch
- Map page in the dropdown menu
- Contributions page for each epoch header
- Allocations page header
- in export csv name
- Distribution Page

## Test and Deployment Plan
Edit and create epochs with description field and check where it is displayed as above

## Screenshots (if appropriate):
image
image
image
image
image
image
image
image

### CODE REVIEW

File: src/pages/AssetMapPage/AMDrawer.tsx
Comment (topocount) [inline src/pages/AssetMapPage/AMDrawer.tsx:137]: this looks like it might be a big ask, but is it possible to string-concat the description with the date range of the epoch?

**Review by topocount — COMMENTED** [review]
This feature is coming together quickly! One question below

File: src/pages/AssetMapPage/AMDrawer.tsx
Comment (sshmm) [reply src/pages/AssetMapPage/AMDrawer.tsx]: done :)

**Comment by crabsinger** [comment]
@sshmm whats status of this? seems its been sitting for a while. what's needed to get this finished?

**Comment by sshmm** [comment]
> @sshmm whats status of this? seems its been sitting for a while. what's needed to get this finished?

I am waiting for reviewers approval since then

**Comment by crabsinger** [comment]
@sshmm ok, can you try to get the CI failures green.

**Comment by levity** [comment]
@sshmm 

Also note:
- on the allocations page, Ivy intentionally [omitted](https://www.figma.com/file/L4oZwYVUdpgacZtwipGaYe?node-id=26298:81159#293249950) the description
- on the contributions page, the description should appear below the date range in normal-sized text ([see here](https://www.figma.com/file/L4oZwYVUdpgacZtwipGaYe?node-id=26309:77557#293249854))

File: src/pages/DistributionsPage/DistributionsPage.tsx
Comment (sshmm) [reply src/pages/DistributionsPage/DistributionsPage.tsx:197]: I wanted to distinguish between epoch number case
image
and description case
image

Comment (levity) [reply src/pages/DistributionsPage/DistributionsPage.tsx:197]: ok, that makes sense

**Comment by levity** [comment]
Pushing this to staging so we can test on the preview app

**Comment by levity** [comment]
Spacing should match

Coordinape

**Comment by levity** [comment]
Spacing should match here

Coordinape

**Review by levity — CHANGES_REQUESTED** [review]
Code looks good, just have some suggestions for visual fixes, see above

**Comment by sshmm** [comment]
> Code looks good, just have some suggestions for visual fixes, see above

I fixed the spacing issues. Please recheck and approve if everything is fine

**Review by levity — APPROVED** [review]

**Pull request merged**
